components/dashboard/src/service/service.tsx

```diff
@@ -111,13 +111,17 @@ export class IDEFrontendService implements IDEFrontendDashboardService.IServer {
 
     private async processServerInfo() {
         this.user = await this.service.server.getLoggedInUser();
-        const workspace = await this.service.server.getWorkspace(this.workspaceID);
-        this.instanceID = workspace.latestInstance?.id;
+        const listener = await this.service.listenToInstance(this.workspaceID);
+
+        const status = this.getWorkspaceStatus(listener.info);
```

Comment:
(nit) it seem it can be generalized and simplified to something like

```ts
const listener = ...
const reconcile = () => {
    ....
}
reconcile();
listener.onDidChange(reconcile);
}